### PR TITLE
Patricksbranch

### PR DESCRIPTION
**Pull request by PatrickRodo**
Read comments please. Some small bugs I mentioned. Finished a lot of the views. Please try to get everything done tomorrow as I will too. Margaret email Ben with the question you had this way we dont have to wait until monday. Denali add validation to the create a menu date. Have it throw the session flash error that is already implemented if a user tries to set a date for the past. Have it check to make sure the date is of today or in the future. I've coded for 17 hrs today so I'll be sleeping until the afternoon. I still need to redesign the find a table page when denalis pull request is merged. Small redesign changes left for edit/create profile views. I didn't get to checking the approve/deny messages. I'll do that later today. Also not sure if you got the links to users profiles working on the find a table page denali. I just havent checked your pull request, but if that functionality still isnt in place get it done today. I'm sure I'm forgetting to mention something. I'll bring anything else up I remember after I go home and sleep.

### CODE REVIEW

**Comment by PatrickRodo** [comment]
I know merge cant be made because I'm not up to date w master. I'm too tired to go through merge conflicts right now I'll do it after I sleep.

**Comment by PatrickRodo** [comment]
Tested Messaging & request a seat:

1.Logged in as Patrick
2.Requested a seat @ Gloria's table.
3.Check Patrick's messages and sees he has a message showing that he requested a seat @ Gloria's table. He can click on message to view its contents. He has access to the approve or deny buttons (Bug. Guest should not be able to see buttons..only host)
4.Logged out as Patrick
5.Logged in as Gloria
6.Check messages
7.Message is available showing Patrick requested a seat at specific menu
8.Can see accept or deny buttons after clicking on message to view contents
9.Click accept and an error is thrown "Creating default object from empty value" (Bug)
10.Click deny and an error is thrown "Creating default object from empty value" (Bug)

Total of 3 bugs. 

We're almost done guys, keep up the hustle we've had the past couple of days.

**Comment by PatrickRodo** [comment]
Not going to style the approve or deny requests because I saw what you were saying about being clever with them. I'm not sure if they're going to be moved into the view instead and given other tags so please let me know as soon as you get them working so I can finalize styling.

File: app/models/Menu.php [hunk 1]
```diff
@@ -17,16 +17,6 @@ class Menu extends BaseModel {
         'cuisine'     => 'required'
     ];
```
Comment (MoseyM) [inline app/models/Menu.php]: Why are the messages gone?

**Comment by PatrickRodo** [comment]
Found one more bug. If a menu has guests you can't delete that menu. Editing it is still possible. 

Error:"Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`capstone_db`.`guests`, CONSTRAINT `guests_menu_id_foreign` FOREIGN KEY (`menu_id`) REFERENCES `menus` (`id`)) (SQL: delete from `menus` where `id` = 11)"

Guest & messages tables are dependent on that menu. Quickest work around I can think of is to shoot off a session flash that says you can't delete a menu that already has guests. This problem would require us to completely restructure things with either the database or controllers and we don't have time for that.

File: app/models/Menu.php (hunk 1 above)
Comment (PatrickRodo) [inline app/models/Menu.php]: Mentioned it in the commit message..

"Removed individual errors. Impossible to have the design that I have and include those errors. It throws everything off. I instead used a session flash. Still leaves for a good UX"

File: app/views/profile/profile.blade.php [hunk 2]
```diff
@@ -25,9 +25,11 @@
                 @else
                 <img class="img-responsive" src="/images/default_profile.jpg" alt="">
                 @endif
-                <input id="input-id" type="number" class="rating profile-rating" data-size="xs">
 				<p>
+<<<<<<< HEAD
```
Comment (MoseyM) [inline app/views/profile/profile.blade.php]: REmove this

**Comment by MoseyM** [comment]
Regarding the deleting of menus. SO what are we trying to do??? Allow the host to delete a menu whether or not guests has signed up? 
**We shouldn't use error msgs here. What is the goal?

**Comment by PatrickRodo** [comment]
GOAL: Host should be able to delete menus whether they have guests for it or not. 

Right now they can only delete menus if they don't have any guests. Say they need to cancel that nights event but they already have guests signed up. They should be able to delete the menu and everything associated with it (guest relationships). If this requires restructuring we don't have time for it. There's still bugs on messages as well as create profile view(can't save profile and background pictures unless you're in the edit profile view)

**Comment by MoseyM** [comment]
google this to get a better understanding of foreign keys or ask ben. plenty of time and very important. 
ie http://laravel.com/docs/4.2/schema#foreign-keys

if im wrong im wrong but i doubt it.

File: app/controllers/GuestsController.php [hunk 3]
```diff
@@ -24,13 +24,13 @@ protected function createNotification(Guest $guest, Menu $menu)
 	{
 		$requestor = User::where('id', $guest->guest_id)->first();
         $conversation = Conversation::create([
-            'subject' => "$requestor->firstname Has Requested a Seat for $menu->title",
+            'subject' => "$requestor->firstname has requested a seat for \"$menu->title\"",
```
Comment (MoseyM) [inline app/controllers/GuestsController.php]: Why is this being edited. There is no need to escape or interpolate the menu title.

Comment (PatrickRodo) [inline app/controllers/GuestsController.php]: Added quotes around the menu title when it's being called in the conversation view, so yes I do need to escape it

**Pull request merged**